packages/ethereum/contracts/HoprNetworkRegistry.sol

```diff
@@ -0,0 +1,146 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.0;
+
+import '@openzeppelin/contracts/access/Ownable.sol';
+import './IHoprNetworkRegistryRequirement.sol';
+
+/**
+ * @dev Allowlist addresses and hopr nodes
+ */
+contract HoprNetworkRegistry is Ownable {
+  IHoprNetworkRegistryRequirement public requirementImplementation;
+  mapping(address => string) public accountToNodeMultiaddr;
+  mapping(string => address) public nodeMultiaddrToAccount;
+
+  event RequirementUpdated(address indexed requirementImplementation);
+  event Registered(address indexed account, string HoprMultiaddr);
+  event RegisteredByOwner(address indexed account, string HoprMultiaddr);
```

Comment:
Wouldn't it make more sense to just use one event, disregarding who registered the node? Then we would only have to listen to one event each and not two, simplifying things a bit.

Reply:
The indexer anyway listen to all the events and filter them. Splitting those two allows node to distinguish registered peer id and may (in the future) provide us more options on treating nodes according to their nature